core/array.rb

```diff
@@ -1516,30 +1516,13 @@ def sample(count=undefined, options=undefined)
         while i < count
           k = rng.rand(size)
 
-          spin = false
-          spin_count = 0
-
-          while true
```

Comment:
Once `spin` is set to true, `spin` will never become false in this loop.
In that case, `spin_count` will always becomes over 100, and always abandoned.
(so I removed `spin`, `spin_count`, and this `while true` loop)